src/prefect/utilities/processutils.py

```diff
@@ -55,6 +56,7 @@ async def run_process(
     command: List[str],
     stream_output: Union[bool, Tuple[Optional[TextSink], Optional[TextSink]]] = False,
     task_status: Optional[anyio.abc.TaskStatus] = None,
+    started_callback: Optional[Callable[[anyio.abc.Process], None]] = None,
```

Comment:
Why were these changes necessary? Now we can't use `TaskGroup.start` with this utility.

Reply:
Because I wanted the hostname to be part of the `pid` sent back to the Agent, but I also wanted `Process` to be in charge of how that was formatted.

Reply:
Ah I see. Let's just introduce a new task group in the `Process` then i.e.

```python
async with anyio.create_task_group() as tg:
   pid = tg.start(run_process, ....)
   task_status.started(f"{hostname}:{pid}")
```

Reply:
I tried this route, but since `run` needs to block on waiting for the process starting up a new task group was untenable. I went with a modified approach of my initial commit that allows for a 'handler' for what's sent back via task_status.

Reply:
Ah it'll block until it's complete (when the task group context exits) but yeah it's a pain to get the return value (`process`) with task groups without introducing a new utility. This seems reasonable for now, thanks for looking into it!